scripts/get_mpi_implementation.cmake

```diff
@@ -0,0 +1,58 @@
+# Detect MPI implementation. If an implementation is detected, MPI_IMPL is set with "openmpi", "mpich" or "intel". Use
+# the global variable IS_CONDA
+function(get_mpi_implementation)
+  # On standard environment, we search implementation names in the MPI_LIBRARIES variable provided by find_mpi. Inside
+  # conda environment, the library path is the same for all libraries. So we use the 'mpi' variable that is defined
+  # (only) in the build environment. Outside the build environment, the 'mpi' variable is not defined, we use the list
+  # of installed packages. It works because, the mpi packages are in conflict (only one implementation on the same
+  # environment)
+  if(IS_CONDA)
+    if(DEFINED ENV{mpi})
+      set(DETECTION_MESSAGE "from conda build environment")
+      set(VAR_CONDA_MPI $ENV{mpi})
+    else()
+      set(DETECTION_MESSAGE "from conda standard environment")
+      execute_process(
+        COMMAND conda list
+        COMMAND grep mpi
+        OUTPUT_VARIABLE VAR_CONDA_MPI)
+    endif()
+  endif()
+
+  # Outside conda or inside conda without mpi packages, we use the path given by find_mpi
```

Comment:
Thus, one authorizes that the Conda package uses external MPI implementation? Is this fallback to system-wide MPI the desired behaviour? Shouldn't we issue `FATAL_ERROR` directly instead of backing off to system-wide MPI?